### DISCUSSION TITLE
How to use Nginx to reverse proxy docker containers running code-server?

### DISCUSSION THREAD

**Discussion by ghost** [coder/code-server#6730] [Q&A]
I'm running code-server with a docker container. If there's only one container, I used the follow configuration directives:
`proxy_pass http://ip:port;`
`proxy_http_version 1.1;`
`proxy_set_header Host $http_host;`
`proxy_set_header Upgrade $http_upgrade;`
`proxy_set_header Connection upgrade;`
And it works normally.
But if I have many containers, and each has its own IP and Port, I used the follow configuration directives:
`proxy_pass http://$http_x_ip:$http_x_port;`
`proxy_http_version 1.1;`
`proxy_set_header Host $http_host;`
`proxy_set_header Upgrade $http_upgrade;`
`proxy_set_header Connection upgrade;`
I passed IP and Port in request header, and it can access the code-server but with such an error:
`
"An unexpected error occurred that requires a reload of this page. 
The workbench failed to connect to the server (Error: WebSocket close with status code 1006)"
`
So, what can I do to fix it?
Thanks a lot!

**Comment by eliluong** [comment]
You might need to also add `proxy_set_header Accept-Encoding gzip;`

Here is thread with lots of nginx-related information: https://github.com/coder/code-server/issues/4443
